packages/bp/src/core/messaging/messaging-router.ts

```diff
@@ -19,10 +19,14 @@ export class MessagingRouter extends CustomRouter {
     this.router.post(
       '/receive',
       this.asyncMiddleware(async (req, res, next) => {
-        if (req.headers.password !== process.INTERNAL_PASSWORD) {
+        if (!this.messaging.isExternal && req.headers.password !== this.messaging.internalPassword) {
           return next?.(new UnauthorizedError('Password is missing or invalid'))
         }
 
+        if (req.body?.type === 'health') {
+          return res.sendStatus(200)
+        }
```

Comment:
Can you sync this PR with messaging head? Those fixes to work with the health service are already in that branch

Reply:
I'll automatically be done once I merge this PR. The only reason why you are seeing this change here is because my base branch isn't up-to-date with sm-messaging-head!